### PR TITLE
[plugin] use namespace to get the PodDisruptionBudgetList

### LINKED ISSUE #4522: [Bug]: Kubectl cnpg status forbidden for non cluster-admins
### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

_No response_

### Version

older in 1.22.x

### What version of Kubernetes are you using?

1.28

### What is your Kubernetes environment?

Other

### How did you install the operator?

Helm

### What happened?

Kubectl pulgin version:  v1.23.1

When using "kubectl cnpg status" as namespaced admin the following error occured
```
kubectl cnpg status -n db-test db-test  --as-group namespace-admin --as foobar
Error: while extracting PodDisruptionBudgetList: poddisruptionbudgets.policy is forbidden: User "foobar" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
```
and no status is printed
Maybe a regression of https://github.com/cloudnative-pg/cloudnative-pg/pull/4319

### Cluster resource

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by HaveFun83**
This PR mitigate the behavior of `kubectl cnpg status` when the user don't have sufficient rights to list  poddisruptionbudgets at the cluster scrope an error and the cnpg status output will be displayed. 

example output
```bash
❯ kubectl cnpg status -n db-test db-test  --as-group namespace-admin --as foobar
poddisruptionbudgets.policy is forbidden: User "foobar" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
Cluster Summary
Name:                db-test
Namespace:           db-test
System ID:           7237111851009441825
PostgreSQL Image:    cloudnative-pg/postgresql:14.8-4
Primary instance:    db-test-1
Primary start time:  2024-04-25 16:11:26 +0000 UTC (uptime 451h23m46s)
Status:              Cluster in healthy state 
Instances:           2
Ready instances:     2
Current Write LSN:   0/31000000 (Timeline: 11 - WAL File: 0000000B0000000000000030)
<omit>
```

Fixes #4522

### CODE REVIEW

**Comment by HaveFun83** [comment]
i think b4c24da5c02120da19537815c7cf6026c8457e8e  fixed it in a better way as the namespace is mandatory to get the cluster status. Now also non cluster-admin users are able to receive the correct pdb status.

**Comment by smartbit** [comment]
> **github-actions** added backport-requested ◀️[ release-1.21 ](https://github.com/cloudnative-pg/cloudnative-pg/labels/release-1.21) ◀️[release-1.22](https://github.com/cloudnative-pg/cloudnative-pg/labels/release-1.22) ◀️[ release-1.23](https://github.com/cloudnative-pg/cloudnative-pg/labels/release-1.23)

As these lines are added to `internal/cmd/plugin/status/status.go` in [1.22.2..1.22.3](https://github.com/cloudnative-pg/cloudnative-pg/compare/v1.22.2...v1.22.3#diff-6a10da9d98890826e17b3c0795a4b9a99f9c63d32dd9d6194e2b3c83d4a14f7d)
```go
157 var pdbl policyv1.PodDisruptionBudgetList
158 if err := plugin.Client.List(ctx, &pdbl, client.MatchingLabels{utils.ClusterLabelName: clusterName}); err != nil {
159		return nil, fmt.Errorf("while extracting PodDisruptionBudgetList: %w", err)
160 }
```
it can be backported to 
- `1.22.3`
- `1.23.0`
- `1.23.1`

**Comment by leonardoce** [comment]
Hi @HaveFun83, thank you for contributing to CNPG!
Can you please fix the issues raised by the linter?

**Comment by HaveFun83** [comment]
sure i will try

**Comment by leonardoce** [comment]
Started an E2e test run on the EDB repo to save some OSS workers: https://github.com/EnterpriseDB/cloudnative-pg/actions/runs/9094968817

**Comment by HaveFun83** [comment]
@leonardoce thanks a lot for your

**Comment by leonardoce** [comment]
E2e tests are green

**Review by leonardoce — APPROVED** [review]

**Comment by leonardoce** [comment]
Looks good to me. Waiting for another maintainer to review it.

**Review by mnencia — APPROVED** [review]

**Pull request merged**
